otcextensions/sdk/cloud/dds.py

```diff
@@ -0,0 +1,190 @@
+                network, ignore_missing=False)
+            attrs['subnet_id'] = network_obj.id
+
+        if security_group:
+            security_group_obj = self.network.find_security_group(
+                security_group, ignore_missing=False)
+            attrs['security_group_id'] = security_group_obj.id
+
+        if password:
+            attrs['password'] = password
+
+        if disk_encryption_id:
+            attrs['disk_encryption_id'] = disk_encryption_id
+
+        if mode:
+            if mode not in ['Sharding', 'ReplicaSet']:
+                raise exceptions.SDKException(
+                    '`Sharding` or `ReplicaSet` are supported values'
+                )
+            attrs['mode'] = mode
+
+        flavors_ref = list(self.dds.flavors(
+            region=region,
+            engine_name=datastore_type)
+        )
+        flavors_specs = [flavor.spec_code for flavor in flavors_ref]
+        if flavors:
+            for flavor in flavors:
+                if flavor['type'] in ['mongos', 'shard'] \
+                        and flavor['num'] not in range(2, 16):
+                    raise exceptions.SDKException(
+                        '`num` value must be in ranges from 2 to 16 '
+                        'for mongos and shard'
+                    )
+                if flavor['type'] == ['config', 'replica'] \
+                        and flavor['num'] != 1:
+                    raise exceptions.SDKException(
+                        '`num` value must be 1 '
+                        'for config and replica'
+                    )
+                if flavor['type'] == 'mongos':
+                    if all(k in flavor for k in ('storage', 'size')):
+                        raise exceptions.SDKException(
+                            '`storage` and `size` parameters'
+                            ' is invalid for the mongos nodes'
+                        )
+                if 'size' in flavor:
+                    if flavor['type'] == 'replica' \
+                            and not (10 <= flavor['size'] <= 2000):
+                        raise exceptions.SDKException(
+                            '`size` value for `replica` must be'
+                            ' between 10 and 2000 GB.'
+                        )
+                    elif flavor['type'] == 'config' \
+                            and flavor['size'] != 20:
+                        raise exceptions.SDKException(
+                            '`size` value for `config` must be 20 GB.'
+                        )
+                    elif not (10 <= flavor['size'] <= 1000):
+                        raise exceptions.SDKException(
+                            '`size` value for `shard` must be'
+                            ' between 10 and 1000 GB.'
+                        )
+                if flavor['spec_code'] not in flavors_specs:
+                    raise exceptions.SDKException(
+                        '`spec_code` not valid'
+                    )
```

Comment:
Are you sure you want to raise an exception if one flavor in the list has invalid field?